Lib/test/test_configparser.py

```diff
@@ -720,6 +721,11 @@ def test_read_returns_file_list(self):
         parsed_files = cf.read(file1)
         self.assertEqual(parsed_files, [file1])
         self.assertEqual(cf.get("Foo Bar", "foo"), "newbar")
+        # check when we pass only a Path object:
+        cf = self.newconfig()
+        parsed_files = cf.read(pathlib.Path(file1))
+        self.assertEqual(parsed_files, [pathlib.Path(file1)])
```

Comment:
Those methods also correctly handle bytestring paths. If you change line 700 to use os.fspath() and attempt to use a bytestring here, it will fail if one of the byte values for a character matches a file handle:

```
ConfigParser().read(b'\x03')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/david/develop/cpython/Lib/configparser.py", line 700, in read
    read_ok.append(os.fspath(filename))
TypeError: expected str, bytes or os.PathLike object, not int
```